client/src/main/java/io/pravega/client/stream/impl/EventPointerImpl.java

```diff
@@ -52,4 +61,62 @@ int getEventLength() {
     public EventPointerInternal asImpl() {
         return this;
     }
+    
+    public static EventPointer fromString(String eventPointer) {
+        int i = eventPointer.lastIndexOf(":");
+        Preconditions.checkArgument(i > 0, "Invalid event pointer: %s", eventPointer);
+        String[] offset = eventPointer.substring(i + 1).split("-");
+        Preconditions.checkArgument(offset.length == 2, "Invalid event pointer: %s", eventPointer);
+        return new EventPointerImpl(Segment.fromScopedName(eventPointer.substring(0, i)), Long.parseLong(offset[0]),
+                                    Integer.parseInt(offset[1]));
+    }
+    
+    @Override
+    public String toString() {
+        return segment.getScopedName() + ":" + eventStartOffset + "-" + eventLength;
```

Comment:
Here you use string concat, yet somewhere above you use `StringBuffer`. I would like it if we were consistent.